src/v/raft/consensus.cc

```diff
@@ -1447,11 +1448,29 @@ ss::future<> consensus::do_start() {
             _vote_timeout.rearm(next_election);
         }
 
-        auto last_applied = read_last_applied();
-        if (last_applied > _commit_index) {
-            _commit_index = last_applied;
-            maybe_update_last_visible_index(_commit_index);
-            vlog(_ctxlog.trace, "Recovered commit_index: {}", _commit_index);
+        auto const last_applied = read_last_applied();
+        if (last_applied > lstats.dirty_offset) {
+            vlog(
+              _ctxlog.error,
+              "Inconsistency detected between KVStore last_applied offset({}) "
+              "and log end offset ({}).  If the storage directory was not "
+              "modified intentionally, this is a bug. Raft in initial state: "
+              "{}",
```

Comment:
>               "and log end offset ({}).  If the storage directory was not "
              "modified intentionally, this is a bug. Raft in initial state: "

I'm confused about what this message means. Is there a bug in the code? If so, then would we fix it? If there isn't a bug why would we say there might be one? Further, is it always safe to handle the inconsistency like we are doing, or is there some case where we should stop execution and be cautious?

Reply:
this is basically a copy of a message from [controller.cc ](https://github.com/redpanda-data/redpanda/blob/ff83af154e3886e4a00ba4223c070ad0de148bfd/src/v/cluster/controller.cc#L343C19-L350C44), 
We even have a test asserting if that message is printed out when starting with empty controller log. The correct way to handle that case is to remove the `last_applied` offset. This will prevent controller from logging this error. 
@jcsp what do you think about dropping that message completely and allow Raft to handle it ?